pkg/controller/storageversionmigrator/resourceversion.go

```diff
@@ -0,0 +1,272 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storageversionmigrator
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/metadata"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/controller"
+
+	svmv1alpha1 "k8s.io/api/storagemigration/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	svminformers "k8s.io/client-go/informers/storagemigration/v1alpha1"
+	clientset "k8s.io/client-go/kubernetes"
+	svmlisters "k8s.io/client-go/listers/storagemigration/v1alpha1"
+)
+
+const (
+	rvAddedConditionType svmv1alpha1.MigrationConditionType = "ResourceVersionAdded"
```

Comment:
Can't we tell whether the RV is added by doing a len check on the .status.RV?  If so, why bother with a condition?

Reply:
makes sense. removed.